src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -313,13 +314,19 @@ public IncomingReplicationHandler CreateIncomingInstance(TcpConnectionOptions tc
         private ReplicationLatestEtagRequest IncomingInitialHandshake(TcpConnectionOptions tcpConnectionOptions, JsonOperationContext.ManagedPinnedBuffer buffer)
         {
             ReplicationLatestEtagRequest getLatestEtagMessage;
+
             using (tcpConnectionOptions.ContextPool.AllocateOperationContext(out JsonOperationContext context))
             using (var readerObject = context.ParseToMemory(
                 tcpConnectionOptions.Stream,
                 "IncomingReplication/get-last-etag-message read",
                 BlittableJsonDocumentBuilder.UsageMode.None,
                 buffer))
             {
+
+                var exceptionSchema = JsonDeserializationClient.ExceptionSchema(readerObject);
+                if (exceptionSchema.Type.Equals("Error"))
+                    throw new Exception(exceptionSchema.Message);
```

Comment:
Why not create an overload for ExceptionDispatcher.Throw to which we are passing the schema and handling that correctly?

Reply:
ExceptionDispatcher expect a different format.. we will be handling on unify those here:
https://issues.hibernatingrhinos.com/issue/RavenDB-15455